packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java

```diff
@@ -183,37 +200,37 @@ private void setup(
       final Activity activity,
       final PluginRegistry.Registrar registrar,
       final ActivityPluginBinding activityBinding) {
-    this.activity = activity;
     this.application = application;
-    this.delegate = constructDelegate(activity);
+    delegate = constructDelegate(activity);
+    activityState = new ActivityState(activity);
     channel = new MethodChannel(messenger, CHANNEL);
     channel.setMethodCallHandler(this);
-    observer = new LifeCycleObserver(activity);
     if (registrar != null) {
       // V1 embedding setup for activity listeners.
-      application.registerActivityLifecycleCallbacks(observer);
+      application.registerActivityLifecycleCallbacks(activityState.getObserver());
       registrar.addActivityResultListener(delegate);
       registrar.addRequestPermissionsResultListener(delegate);
     } else {
       // V2 embedding setup for activity listeners.
       activityBinding.addActivityResultListener(delegate);
       activityBinding.addRequestPermissionsResultListener(delegate);
       lifecycle = FlutterLifecycleAdapter.getActivityLifecycle(activityBinding);
-      lifecycle.addObserver(observer);
+      lifecycle.addObserver(activityState.getObserver());
     }
   }
 
   private void tearDown() {
     activityBinding.removeActivityResultListener(delegate);
     activityBinding.removeRequestPermissionsResultListener(delegate);
     activityBinding = null;
-    lifecycle.removeObserver(observer);
+    lifecycle.removeObserver(activityState.getObserver());
     lifecycle = null;
     delegate = null;
     channel.setMethodCallHandler(null);
     channel = null;
-    application.unregisterActivityLifecycleCallbacks(observer);
+    application.unregisterActivityLifecycleCallbacks(activityState.getObserver());
     application = null;
+    activityState = null;
   }
```

Comment:
Isn't everything that's created and destroyed in these two methods by definition activity state? It's not clear to me why only these two things are being moved.

Reply:
Yes. Please have another look at this PR, Thanks.